### PR TITLE
 deploy: update templates to v3.12.1

### LINKED ISSUE #4777: ceph-csi 3.12.0 not able to provision RBD PVC.
> @iPraveenParihar can you please check above?

https://github.com/ceph/ceph-csi/pull/4776 should fix the `CSI NodePlugin with v.3.12.0 driver fails when missing domain labels` deployed via helm.

> this does indeed "fixes" the mentioned problem, however, on new pvc, there's new one :D

We yet to decide on how to enable/disable topology aware provisioning (https://github.com/ceph/ceph-csi/pull/4726).
For now, the workaround is to add `--feature-gates=Topology=false` for csi-provisioner in csi-rbdplugin-provisioner deployment pod.

@ypyly, Can you please try with above workaround and see if that helps you?

_Originally posted by @iPraveenParihar in https://github.com/ceph/ceph-csi/issues/4775#issuecomment-2295910213_

### PR DESCRIPTION
**Pull request by Rakshith-R**
This commit updates templates to new
v3.12.1 release.

Fixes: #4777 

**Checklist:**

* [ ] **Commit Message Formatting**: Commit titles and messages follow
  guidelines in the [developer
  guide](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#commit-messages).
* [ ] Reviewed the developer guide on [Submitting a Pull
  Request](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#development-workflow)
* [ ] [Pending release
  notes](https://github.com/ceph/ceph-csi/blob/devel/PendingReleaseNotes.md)
  updated with breaking and/or notable changes for the next major release.
* [ ] Documentation has been updated, if necessary.
* [ ] Unit tests have been added, if necessary.
* [ ] Integration tests have been added, if necessary.

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

* `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)

</details>

### CODE REVIEW

File: build.env [hunk 1]
```diff
@@ -9,7 +9,7 @@
 # get proporly expanded.
 #
 # cephcsi image version
-CSI_IMAGE_VERSION=v3.12-canary
+CSI_IMAGE_VERSION=v3.12.1
```
Comment (Madhu-1) [inline build.env:12]: can you please update the version in the upgrade doc as well?

File: docs/ceph-csi-upgrade.md [hunk 2]
```diff
@@ -98,12 +98,12 @@ For example, upgrading from 3.10 to 3.12 is not recommended.
 [release notes](https://github.com/ceph/ceph-csi/releases/tag/v3.12.0) before
```
Comment (Madhu-1) [inline docs/ceph-csi-upgrade.md:98]: update the tag above as well? It pointing to 3.12.0

Comment (Rakshith-R) [reply docs/ceph-csi-upgrade.md:98]: > update the tag above as well? It pointing to 3.12.0

That refers to breaking changes in release notes. 
we usually have breaking changes only in major version 0th release.
hence, left it as it is.
Let me know if you still want to update it.

We refer to 3.10.0 release notes for breaking changes here too

https://github.com/ceph/ceph-csi/blob/release-v3.10/docs/ceph-csi-upgrade.md#upgrading-from-v39-to-v310

**Review by Madhu-1 — APPROVED** [review]

File: build.env [hunk 3]
```diff
@@ -9,7 +9,7 @@
...
 
 # cephcsi upgrade version
 CSI_UPGRADE_VERSION=v3.11.0
```
Comment (iPraveenParihar) [inline build.env:15]: we are not updating `CSI_UPGRADE_VERSION` to latest?

Comment (Rakshith-R) [reply build.env:15]: That needs to be done in devel branch, not release branch.

**Review by iPraveenParihar — APPROVED** [review]

**Comment by Rakshith-R** [comment]
@mergifyio queue

File: build.env (hunk 3 above)
Comment (iPraveenParihar) [reply build.env:15]: https://github.com/ceph/ceph-csi/pull/4785

**Pull request merged**